disttask/importinto/encode_and_sort_operator.go

```diff
@@ -116,6 +126,40 @@ func (op *encodeAndSortOperator) Done() <-chan struct{} {
 type chunkWorker struct {
 	ctx context.Context
 	op  *encodeAndSortOperator
+
+	dataWriter  *external.EngineWriter
+	indexWriter *importer.IndexRouteWriter
+}
+
+func newChunkWorker(ctx context.Context, op *encodeAndSortOperator, workerID int64) *chunkWorker {
+	w := &chunkWorker{
+		ctx: ctx,
+		op:  op,
+	}
+	if op.tableImporter.IsGlobalSort() {
+		// sorted index kv storage path: /{taskID}/{subtaskID}/index/{indexID}/{workerID}
+		indexWriterFn := func(indexID int64) *external.Writer {
+			builder := external.NewWriterBuilder().
+				SetOnCloseFunc(func(summary *external.WriterSummary) {
+					op.sharedVars.mergeIndexSummary(indexID, summary)
+				})
+			prefix := path.Join(strconv.Itoa(int(op.taskID)), strconv.Itoa(int(op.subtaskID)))
+			writerID := path.Join("index", strconv.Itoa(int(indexID)), strconv.Itoa(int(workerID)))
```

Comment:
Considering two nodes are running same subtask (maybe the old one suffers network partition and framework dispatch a new node), we should use random writerID to avoid overwritten. Or maybe WriterBuilder should assign UUID internally 🤔 

The result file path will be returned by OnClose, so caller seems don't need to care about the writerID

Reply:
cc @wjhuang2016 @tangenta

Reply:
> Considering two nodes are running same subtask

makes sense, will change it to a uuid here

> so caller seems don't need to care about the writerID

that would be better

Reply:
https://github.com/pingcap/tidb/pull/46998/commits/a3e9817aaf3f5987c3ca20965bf63682096ed59c